addons/pos_discount/static/src/js/DiscountButton.js

```diff
@@ -37,31 +37,40 @@ odoo.define('pos_discount.DiscountButton', function(require) {
             }
 
             // Remove existing discounts
-            for (const line of lines) {
-                if (line.get_product() === product) {
-                    order.remove_orderline(line);
-                }
-            }
+            lines.filter(line => line.get_product().default_code === 'DISC')
```

Comment:
Hello, her you should not use default code to see if it is a discount product, because it can be changed.
compare it with var product which is the discount product defined on the POS (this.env.pos.config.discount_product_id[0])

Otherwise, everything else seems good

Reply:
Hi @pimodoo, do you mean something like:
`lines.filter(line => line.get_product() === product)`
where product was declared as:
`var product  = this.env.pos.db.get_product_by_id(this.env.pos.config.discount_product_id[0]);`
Is that right? :)

Reply:
yes

Reply:
Great, thanks for the review!! :D